Author: Michael Vogt <michael.vogt@ubuntu.com>
Description: map windows with focus_on_map=FALSE in the background 
 map windows with focus_on_map=FALSE in the background (just like metacity)
Origin: vendor, ubuntu (1:0.8.2-0ubuntu6)
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/compiz/+bug/333284
--- compiz.orig/src/event.c
+++ compiz/src/event.c
@@ -2073,7 +2073,7 @@ handleEvent (CompDisplay *d,
 		focus = allowWindowFocus (w, NO_FOCUS_MASK,
 					  w->screen->x, w->screen->y, 0);
 
-		if (focus == CompFocusDenied)
+		if ((focus == CompFocusDenied || focus == CompFocusPrevent) && (w->type & ~NO_FOCUS_MASK))
 		    stackingMode = CompStackingUpdateModeInitialMapDeniedFocus;
 		else
 		    stackingMode = CompStackingUpdateModeInitialMap;
